File: samples/hello_ice_time.c
/* Define the implementation of the library and include it */
#define ICE_TIME_IMPL 1
#include "ice_time.h"

#include <stdio.h>

/* Helper */
#define trace(fname, str) printf("[%s : line %d] %s() => %s\n", __FILE__, __LINE__, fname, str);

int main(void) {
    /* Struct that contains Time information */
    ice_time_info current_time;
    
    /* Fetch time information and store it in the struct */
    ice_time_error res = ice_time_get_info(&current_time);

    /* If the function failed to fetch time information, Trace error then terminate the program! */
    if (res != ICE_TIME_ERROR_OK) {
        trace("ice_time_get_info", "ERROR: failed to get time info!");
        return -1;
    }

    /* Print current time! */
    printf("Current Time: %s\n", current_time.str);

    return 0;
}

File: bindings/luajit/examples/hello_ice_time.lua
local ffi = require("ffi")
local lib = require("ice_time")

-- This allows the functions of the bindings to be globally called...
setmetatable(_G, { __index = lib })

local function main()
  -- Struct that contains Time information
  local current_time = ffi.new("ice_time_info")
  
  -- Fetch time information and store it in the struct
  local res = ice_time_get_info(current_time)
  
  -- If the function failed to fetch time information, Trace error then terminate the program!
  if (res ~= ICE_TIME_ERROR_OK) then
    print("ERROR: failed to retrieve time info!")
    return -1
  end
  
  -- Print current time!
  print("Current Time: " .. ffi.string(current_time.str))
  
  return 0
end

return main()
